belief_manager.py

```diff
@@ -0,0 +1,578 @@
+    self._observed_transmitters = collections.deque()
+    # Error regions calculated from our last cycle.
+    self._last_error_regions = {}
+    # A dictionary of transmitter indices. Each value is the cycle we last got
+    # data for it.
+    self._cycle_data = {}
+
+    # How many cycles we've run.
+    self._cycles = 0
+
+  def _fetch_autopilot_data(self):
+    """ Fetches measurements from the onboard Pixhawk autopilot, and updates the
+    member variables for vehicle position and velocity observations accordingly.
+    """
+    # Read the latest data from the autopilot.
+    sensor_data = self.__autopilot.read_front()
+
+    self._observed_position_x = sensor_data.latitude
+    self._observed_position_y = sensor_data.longitude
+
+    self._observed_velocity_x = sensor_data.velocity[self._X]
+    self._observed_velocity_y = sensor_data.velocity[self._Y]
+
+    logger.info("Got new sensor data: position: (%f, %f), velocity: (%f, %f)" \
+                % (self._observed_position_x, self._observed_position_y,
+                   self._observed_velocity_x, self._observed_velocity_y))
+
+  def _fetch_radio_data(self):
+    """ Fetches measurements from the radio.
+    Returns:
+        A list containing one tuple for every object sighted by the radar.
+        Each tuple contains the LOB measurement and signal strength. """
+    transmitters = []
+
+    # Keep reading data until we have no more to read.
+    while True:
+      signal = self.__radio.read_next(block=False)
+      if not signal:
+        # No more data.
+        break
+
+      # Normalize the LOB, so that the angle will still be correct even if the
+      # drone starts flying in a different direction.
+      nomalized_lob = self._filter.normalize_lobs(signal.lob)
+
+      transmitters.append((normalized_lob, signal.strength))
+
+    return transmitters
+
+  def _associate_lob_readings(self, readings):
+    """ Takes a set of LOB readings from the radio system, and associates them
+    with known transmitters.
+    Args:
+      readings: The set of readings to associate. Should be a list of tuples of
+      the form (LOB, strength)
+    Returns: A dictionary. The keys are indexes of transmitters in the state,
+      and the values are the tuple of the associated LOB and signal strength. It
+      also returns a list of readings that are asumed to be from new
+      transmitters. """
+    margins_of_error = \
+        self._filter.lob_confidence_intervals(self.ERROR_REGION_Z_SCORE)
+
+    # Center the interval around our predicted next LOBs.
+    predicted_state = self._filter.estimate_next_state()
+    intervals = []
+    for i in range(4, len(predicted_state)):
+      lob = predicted_state[i]
+      margin_of_error = margins_of_error[i - 4]
+      intervals.append((lob - margin_of_error, lob + margin_of_error))
+
+    logger.debug("LOB confidence intervals: %s" % (intervals))
+
+    # Now, go and check whether our new LOBs fit within them.
+    associations = {}
+    new = []
```

Comment:
Uhh... I don't see why the name needs to change. It's not like "new" is a protected name in Python.